packages/react-core/src/components/Form/FormGroup.tsx

```diff
@@ -41,6 +47,9 @@ export interface FormGroupProps extends Omit<React.HTMLProps<HTMLDivElement>, 'l
   helperTextInvalidIcon?: React.ReactNode;
   /** ID of the included field. It has to be the same for proper working. */
   fieldId: string;
+  /** Object for identifying whether the form group has multiple inputs for a single label, and
+   * whether the multiple inputs are radio inputs or not. */
+  multipleInputs?: MultipleInputsObject;
```

Comment:
In some examples you're passing `multipleInputs` as a boolean, but it's just typed as an object here. I think that either those examples should be updated to pass an empty object or the type here should be updated to be an object or a boolean.

If you go the type updating route though I feel like this prop name could use a slight edit.

I do also wonder if (as we've talked about a bit) this API would maybe make more sense with two separate props instead of an object prop that wraps another singular prop. I'm not positive how I feel about this idea though and would love input from others 🙂

Reply:
I definitely understand the concern with the API here. A few ways that we could go about this off the top of my head:

- Update the interface so that `isRadioGroup` is required when passing in `multipleInputs`, and updating examples so that either `multipleInputs={{ isRadioGroup: true }}` or `multipleInputs={{ isRadioGroup: false }}` is passed in (as opposed to *just* `multipleInputs`
- Renaming the prop to something along the lines of `groupOptions` or `groupType`, then adding in another prop to the interface along the lines of `inputAmount: 'single' | 'multiple'` (more similar to your DataList implementation). This would result in a prop looking like `groupOptions={{ inputAmount: 'multiple', isRadioGroup: 'true' }}`. Not personally sure if this would make the most sense, however, since the default behavior is already a single input per form group, and this might require passing in props to retain that default behavior.
- Just breaking things out to two separate props like you mentioned, maybe `hasMultipleInputs` and `isRadioGroup`, and just noting in the `isRadioGroup` description that it should only be passed in when `hasMultipleInputs` is also passed in.

Tagging @nicolethoen as well for input on the above proposals

Reply:
Yeah those are all the direct options I can think of as well, and honestly I don't love any of them. Your current API implementation is probably my favorite of these options.

Is there any way we could automatically determine if a group has multiple inputs rather than requiring the consumer to pass that in as a prop? I feel like that would really be the best approach from the consumer side.

Reply:
We could possibly check whether the `children` prop of the Form Group is an array (more than 1 input will return true), but this could possibly get skewed if any non-input content is rendered within the Form Group, such as helper text or validation text. I tinkered around a little and there's some workaround to that, but it may not be 100% accurate all the time due to there being various variables to consider (if it's a native input element, whether it's a React component, whether that React component *is* an input, etc.)

This is just from some quick tinkering around, though, so I might be missing something more obvious.

Reply:
Yeah that doesn't seem ideal. I have one vague idea taking shape which would involve in some way adding registration behavior of some sort to our input components. If that would even work though it would probably be out of scope for this issue, so for the moment at least I'm going to join in saying that your implementation is slick and I think I like it.